src/main/java/uk/gov/hmcts/divorce/common/event/DraftConditionalOrder.java

```diff
@@ -82,9 +82,14 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDet
         final CaseData data = details.getData();
         data.getConditionalOrder().getConditionalOrderApplicant1Questions().setIsDrafted(YES);
 
+        var state = details.getState();
+        if (!details.getState().equals(ConditionalOrderPending)) {
+            state = ConditionalOrderDrafted;
+        }
+
```

Comment:
It would be nice to have this repeated code as a CaseTask

Reply:
Implementing this, will be in next commit